extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js

```diff
@@ -216,235 +140,257 @@ function tests(name) {
         return ad;
       })).to.be.not.be.rejected;
     });
+  });
 
-    it('should only layout once', () => {
-      return getAd().then(ad => {
-        ad.implementation_.unlayoutCallback();
+  it('should only layout once', () => {
+    return getAd().then(ad => {
+      ad.implementation_.unlayoutCallback();
 
-        const firstLayout = ad.implementation_.layoutCallback();
-        const secondLayout = ad.implementation_.layoutCallback();
-        expect(firstLayout).to.equal(secondLayout);
+      const firstLayout = ad.implementation_.layoutCallback();
+      const secondLayout = ad.implementation_.layoutCallback();
+      expect(firstLayout).to.equal(secondLayout);
 
-        ad.implementation_.unlayoutCallback();
-        const newLayout = ad.implementation_.layoutCallback();
-        expect(newLayout).to.not.equal(secondLayout);
+      ad.implementation_.unlayoutCallback();
+      const newLayout = ad.implementation_.layoutCallback();
+      expect(newLayout).to.not.equal(secondLayout);
+    });
+  });
+
+  describe('get CID', () => {
+    it('should propagete CID to ad iframe', () => {
+      sandbox.stub(adCid, 'getAdCid', () => {
+        return Promise.resolve('sentinel123');
+      });
+      return getAd().then(ad => {
+        const src = ad.firstChild.getAttribute('src');
+        expect(src).to.contain('"clientId":"sentinel123"');
       });
     });
 
-    describe('get CID', () => {
-      it('should propagete CID to ad iframe', () => {
-        sandbox.stub(adCid, 'getAdCid', () => {
-          return Promise.resolve('sentinel123');
-        });
-        return getAd().then(ad => {
-          const src = ad.firstChild.getAttribute('src');
-          expect(src).to.contain('"clientId":"sentinel123"');
-        });
+    it('should proceed w/o CID', () => {
+      sandbox.stub(adCid, 'getAdCid', () => {
+        return Promise.resolve(undefined);
       });
+      return getAd().then(ad => {
+        const src = ad.firstChild.getAttribute('src');
+        expect(src).to.contain('"clientId":null');
+      });
+    });
+  });
 
-      it('should proceed w/o CID', () => {
-        sandbox.stub(adCid, 'getAdCid', () => {
-          return Promise.resolve(undefined);
+  describe('no content handler', () => {
+    it('should display fallback when there is one and hide placeholder', () => {
+      return getAd(undefined, undefined, ad => {
+        const fallback = createElementWithAttributes(document, 'div', {
+          fallback: '',
         });
-        return getAd().then(ad => {
-          const src = ad.firstChild.getAttribute('src');
-          expect(src).to.contain('"clientId":null');
+        const placeholder = createElementWithAttributes(document, 'div', {
+          placeholder: '',
         });
+        ad.appendChild(fallback);
+        ad.appendChild(placeholder);
+        expect(placeholder.classList.contains('amp-hidden')).to.be.false;
```

Comment:
I can do this here. because there will be error:
`AssertionError: object tested must be an array, an object, or a string, but domtokenlist given`